src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -535,6 +545,40 @@ protected void parseCreateField(ParseContext context, int dimension) throws IOEx
         context.path().remove();
     }
 
+    // Verify mapping and return true if it is a "faiss" Index using "sq" encoder
+    // of type "fp16"
+    protected boolean isFaissSQfp16(KNNMethodContext knnMethodContext) {
+        if (knnMethodContext != null) {
+            if (knnMethodContext.getKnnEngine().getName().equals(FAISS_NAME)
+                && knnMethodContext.getMethodComponentContext().getParameters().size() != 0) {
+                Map<String, Object> methodComponentParams = knnMethodContext.getMethodComponentContext().getParameters();
+                if (methodComponentParams.containsKey(METHOD_ENCODER_PARAMETER)) {
+                    MethodComponentContext methodComponentContext = (MethodComponentContext) methodComponentParams.get(
+                        METHOD_ENCODER_PARAMETER
+                    );
+                    if (ENCODER_SQ.equals(methodComponentContext.getName())
```

Comment:
I think this case can be combined with a previous one ` if (methodComponentParams.containsKey(METHOD_ENCODER_PARAMETER)) {`, maybe put it to it's own method if it became hard to read